### PR TITLE
Add `configuration-strategy` property to RedisSessionProperties  to configure `ConfigureRedisAction`

### PR DESCRIPTION
**Pull request by nosan**
see gh-16943

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java [hunk 1]
```diff
@@ -40,6 +40,11 @@
 	 */
 	private RedisFlushMode flushMode = RedisFlushMode.ON_SAVE;
 
+	/**
+	 * Allows specifying a strategy for configuring and validating Redis.
+	 */
+	private RedisAction action = RedisAction.NOTIFY_KEYSPACE_EVENTS;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java]: The action interface is quite generic and allows the user to do anything they want to configure and validate the connection. I don't think "action" is the right term for this as what we set is a "configuration and validation strategy". 

`configurationStrategy` is a bit mouthful but more explicit perhaps?

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java [hunk 2]
```diff
@@ -69,4 +74,29 @@ public void setCleanupCron(String cleanupCron) {
...
+	}
+
+	public void setAction(RedisAction action) {
+		this.action = action;
+	}
+
+	/**
+	 * Allows specifying a strategy for configuring and validating Redis.
+	 */
+	public enum RedisAction {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java]: `ConfigurationStrategy` would be the logical name there but it fees a bit odd so a bit more brainstorming is required I think.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java (hunk 1 above)
Comment (vpavic) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/RedisSessionProperties.java]: Having in mind the purpose of `ConfigureRedisAction`, naming this property `configurationStrategy` would be more appropriate.

**Comment by nosan** [comment]
@snicoll @vpavic 
Thank you for your feedback.

PR has been updated.

**Comment by philwebb** [comment]
Thanks for another contribution @nosan! 

I went back and forth on the property name a few times and ended up with `configure-action` in the end. It's a hard one to name because we ideally want some clues that it relates to configuring the Redis server, but we also want to indicate that it's us doing the configuration and to also somehow link back to the `ConfigureRedisAction` interface. I also managed to mess up the polish commit so there's an [additional one](https://github.com/spring-projects/spring-boot/commit/e0aa90cf32e2ae71885271c5645a8b468e46cbec) outside of the [merge commit](https://github.com/spring-projects/spring-boot/commit/1213654c7473db2458132aec3295cbb898f8ce2e).

@snicoll Let me know if you think the name might still be a problem. We still have time to refine it.

**Comment by philwebb** [comment]
> I'd hold this off for a while because we at the Spring Session side should probably take a closer look at our support for configuring ConfigureRedisAction.

Sorry @vpavic, I missed this comment until it was too late. Hopefully whatever changes you've got in mind won't be too hard to apply to the auto-configuration once they're ready. Is there a Spring Session issue we should be tracking?

**Comment by vpavic** [comment]
No problem @philwebb. I've opened spring-projects/spring-session#1458 to consider doing something about this on Spring Session side.

**Pull request closed without merging**